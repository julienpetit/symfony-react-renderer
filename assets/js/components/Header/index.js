import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from './logo.svg';
import './Header.css';

const Header = ({ isAuthenticated = false }) => (
    <header className="App-header">
        <img src={logo} className="App-logo" alt="logo"/>
        <h1 className="App-title">Welcome to React - Redux boilerplate</h1>

        <NavLink to="/" exact activeClassName='active'>Home</NavLink>

        {isAuthenticated ? (
            <NavLink to="/logout" exact activeClassName='active'>Logout</NavLink>
        ) : (
            <NavLink to="/login" exact activeClassName='active'>Login</NavLink>
        )}

        <NavLink to="/account" exact activeClassName='active'>Account</NavLink>
        <NavLink to="/users" exact activeClassName='active'>Users</NavLink>

    </header>
);

Header.propTypes = {
  isAuthenticated: PropTypes.bool,
};

export default Header;
