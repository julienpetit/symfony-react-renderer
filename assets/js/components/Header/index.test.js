import React from 'react';
import Header from './index';
import { shallow } from 'enzyme';

describe('always render subcomponents', () => {
    const wrapper = shallow(
        <Header isAuthenticated={true} />
    );

    test('always renders header', () => {
        const header = wrapper.find('header');
        expect(header.length).toBe(1);
    });

    test('always renders an h1', () => {
        const h1 = wrapper.find('h1');
        expect(h1.length).toBe(1);
    });

    test('always renders 3 NavLink components', () => {
        const navLinks = wrapper.find('NavLink');
        expect(navLinks.length).toBe(3);
    });
});

test('NavLink change when isAuthenticated is true or false', () => {
    let wrapper;

    wrapper = shallow(
        <Header isAuthenticated={true} />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(
        <Header isAuthenticated={false} />
    );

    expect(wrapper).toMatchSnapshot();
});
