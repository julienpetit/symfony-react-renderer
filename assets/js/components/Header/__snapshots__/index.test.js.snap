// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink change when isAuthenticated is true or false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isAuthenticated={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
/>,
        <h1
          className="App-title"
>
          Welcome to React - Redux boilerplate
</h1>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/logout"
>
          Logout
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/account"
>
          Account
</NavLink>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "App-logo",
          "src": "logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to React - Redux boilerplate",
          "className": "App-title",
        },
        "ref": null,
        "rendered": "Welcome to React - Redux boilerplate",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Logout",
          "exact": true,
          "to": "/logout",
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Account",
          "exact": true,
          "to": "/account",
        },
        "ref": null,
        "rendered": "Account",
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
/>,
          <h1
            className="App-title"
>
            Welcome to React - Redux boilerplate
</h1>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/logout"
>
            Logout
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/account"
>
            Account
</NavLink>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "App-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React - Redux boilerplate",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Welcome to React - Redux boilerplate",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Logout",
            "exact": true,
            "to": "/logout",
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Account",
            "exact": true,
            "to": "/account",
          },
          "ref": null,
          "rendered": "Account",
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NavLink change when isAuthenticated is true or false 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isAuthenticated={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
/>,
        <h1
          className="App-title"
>
          Welcome to React - Redux boilerplate
</h1>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/login"
>
          Login
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/account"
>
          Account
</NavLink>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "App-logo",
          "src": "logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to React - Redux boilerplate",
          "className": "App-title",
        },
        "ref": null,
        "rendered": "Welcome to React - Redux boilerplate",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Login",
          "exact": true,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Account",
          "exact": true,
          "to": "/account",
        },
        "ref": null,
        "rendered": "Account",
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
/>,
          <h1
            className="App-title"
>
            Welcome to React - Redux boilerplate
</h1>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/login"
>
            Login
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/account"
>
            Account
</NavLink>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "App-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React - Redux boilerplate",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Welcome to React - Redux boilerplate",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Login",
            "exact": true,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Account",
            "exact": true,
            "to": "/account",
          },
          "ref": null,
          "rendered": "Account",
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
